pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -115,7 +115,8 @@ func (os *OpenStack) getVolume(diskName string) (volumes.Volume, error) {
 		return volume, err
 	}
 
-	err = volumes.List(sClient, nil).EachPage(func(page pagination.Page) (bool, error) {
+	opts := &volumes.ListOpts{Name: diskName}
+	err = volumes.List(sClient, opts).EachPage(func(page pagination.Page) (bool, error) {
```

Comment:
The Cinder API allows you to provide query parameters in the URL when retrieving a list of volumes. Filtering by [name](https://github.com/openstack/cinder/blob/master/cinder/api/v2/volumes.py#L109-L110) for example allows you to restrict the amount of volumes down considerably. Otherwise, the K8s code will traverse the entire collection of volumes in an OpenStack project to find a match (some project could have 10s of thousands). Filtering by name reduces/eliminates the need for subsequent pagination calls.